pkg/machines/scripts/create_machine.sh

```diff
@@ -19,12 +20,16 @@ handleFailure(){
 }
 
 # prepare virt-install parameters
-COMPARISON=$(awk 'BEGIN{ print "'$STORAGE_SIZE'"<=0 }')
-if [ "$COMPARISON" -eq 1 ]; then
-    # default to no disk if size 0
-    DISK_OPTIONS="none"
+if [ "$SOURCE_TYPE" = "disk_image" ]; then
+    DISK_OPTIONS="$SOURCE,device=disk"
 else
-    DISK_OPTIONS="size=$STORAGE_SIZE,format=qcow2"
+    COMPARISON=$(awk 'BEGIN{ print "'$STORAGE_SIZE'"<=0 }')
+    if [ "$COMPARISON" -eq 1 ]; then
```

Comment:
It's not new code, so no need to change it in this PR, but this is  "eww". `if [ "$STORAGE_SIZE" -le 0 ]` will do just fine.